tests/docs/test_supported_confs.py

```diff
@@ -447,17 +440,16 @@ def test_all_finitediff_state(self, interface, return_type, shots, wire_specs):
         msg = "state\\(wires=\\[0?\\]\\)\\ is\\ not\\ in\\ list"
 
         complex = return_type == "StateVector"
+        # the error is no longer raised!
+        circuit = get_qnode(interface, "finite-diff", return_type, shots, wire_specs)
+        x = get_variable(interface, wire_specs, complex=complex)
 
-        with pytest.raises(ValueError, match=msg):
```

Comment:
Following @albi3ro's suggestion, I removed some dead code from `CircuitGraph`. But then this test started failing.

I realised that we are checking for an error that is raised when calling:
https://github.com/PennyLaneAI/pennylane/blob/66b99db10ba0915ea69ac3641106d5482e4e6d86/pennylane/circuit_graph.py#L509

The error was raised because `CircuitGraph` replaced the measurement containing an empty list as wires for a measurement containing all wires. 

Removing this solved the issue for some of the parametrised tests. The ones failing are the following:
- `jax` + {`StateCost`, `StateVector`, `DensityMatrix`}: TypeError: lax.dynamic_update_slice requires arguments to have the same dtypes, got float64, complex128.
- `torch` + {`StateCost`, `DensityMatrix`}: RuntimeError: Expected isFloatingType(grad.scalar_type()) || (input_is_complex == grad_is_complex) to be true, but got false.  (Could this error message be improved?  If so, please report an enhancement request to PyTorch.)
- `autograd` +`StateVector`: ValueError: cannot reshape array of size 4 into shape (2,1)